### PR TITLE
26 Create the seeds

### LINKED ISSUE #26: Create the seeds
This is a **very important** component of the back-end. 

You will need to setup your tables and fields to be in-line with that data model within the [wiki](https://github.com/kahikatea-2019/MOA/wiki/Data-Model).

Tasks: 
- [x] Create seeds.

### PR DESCRIPTION
**Pull request by MurdocAV**
Seed data completed, ready for a pull request into dev branch.
Closes #26

### CODE REVIEW

**Comment by MurdocAV** [comment]
Create the seeds #26

**Review by bryceknz — CHANGES_REQUESTED** [review]
Please see the existing `server/db` folder which already contains a folder for seeds.

**Comment by MurdocAV** [comment]
Should be consistent with the -- postinstal: "db:seed" -- scripts now

File: server/db/seeds/users.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+exports.seed = function (knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('users').del()
+    .then(function () {
+      // Inserts seed entries
+      return knex('users').insert([
+        {id: 1, name: 'Jane', surname: 'Doe', email: 'janedoe@mail.com', hash: '#', role: 'facilitator'},
```
Comment (chrislaupama) [inline server/db/seeds/users.js]: You will need to change field names here, the data model has been updated

File: server/db/seeds/cohorts.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+exports.seed = function (knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('cohorts').del()
+    .then(function () {
+      // Inserts seed entries
+      return knex('cohorts').insert([
+        {id: 1, name: 'Kahikatea', year: 2019, campus: 'Auckland'}
```
Comment (chrislaupama) [inline server/db/seeds/cohorts.js]: the value for year needs to be a string

**Review by chrislaupama — DISMISSED** [review]
You'll need to make a few changes then we'll be all good to go

**Comment by MurdocAV** [comment]
This branch is ready to be merged into dev, checked again 10:13am

File: server/db/seeds/cohorts.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+exports.seed = function (knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('cohorts').del()
+    .then(function () {
+      // Inserts seed entries
+      return knex('cohorts').insert([
+        { id: 1, name: 'Kahikatea', year: '2019', campus: 'Auckland' }
+      ])
```
Comment (bryceknz) [inline server/db/seeds/cohorts.js:10]: Please add a couple more entries so we can ensure good test coverage.

File: server/db/seeds/exercises.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+exports.seed = function (knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('table_name').del()
```
Comment (bryceknz) [inline server/db/seeds/exercises.js]: I think you might've forgotten to pass the new table name to `knex` 😊

File: server/db/seeds/exercises.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    .then(function () {
+      // Inserts seed entries
+      return knex('table_name').insert([
```
Comment (bryceknz) [inline server/db/seeds/exercises.js]: ^Same as above.

**Comment by MurdocAV** [comment]
All fixed, Thanks Bryce

File: server/db/seeds/statuses.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+exports.seed = function (knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('table_name').del()
```
Comment (bryceknz) [inline server/db/seeds/statuses.js]: Another `table_name` error.

File: server/db/seeds/students.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+exports.seed = function (knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('students').del()
+    .then(function () {
+      // Inserts seed entries
+      return knex('students').insert([
+        { id: 1, user_id: 2, cohort_id: 1 }
+      ])
```
Comment (bryceknz) [inline server/db/seeds/students.js:11]: Please add some more students :)

File: server/db/seeds/exercises.js (hunk 5 above)
Comment (MurdocAV) [reply server/db/seeds/exercises.js]: Sorry, I don't know how i got soo many errors

File: server/db/seeds/students_assessments.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+exports.seed = function (knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('students_assessments').del()
+    .then(function () {
+      // Inserts seed entries
+      return knex('students_assessments').insert([
+        {id: 1, student_id: 1, assessment_id: 1, status_id: 4, evidence: 'evidence' , date_modified: 'date'},
+        {id: 1, student_id: 1, assessment_id: 2, status_id: 3, evidence: 'evidence', date_modified: 'date'},
+        {id: 1, student_id: 1, assessment_id: 3, status_id: 2, evidence: 'evidence', date_modified: 'date'},
+        {id: 1, student_id: 1, assessment_id: 4, status_id: 1, evidence: 'evidence', date_modified: 'date'}
```
Comment (bryceknz) [inline server/db/seeds/students_assessments.js]: I've just merged a PR that changes the data type of `data_modified` to a _datetime_. Please update the seed data to reflect this :-)
You may want to use [Date.now()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now)

File: server/db/seeds/students.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+exports.seed = function (knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('students').del()
+    .then(function () {
+      // Inserts seed entries
+      return knex('students').insert([
+        {id: 1, user_id: 2, cohort_id: 1},
```
Comment (bryceknz) [inline server/db/seeds/students.js:7]: Please make the student ids unique.

File: server/db/seeds/exercises.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
+exports.seed = function (knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('exercises').del()
+    .then(function () {
+      // Inserts seed entries
+      return knex('exercises').insert([
+        {id: 1, assessment_id: 1, titie: 'taxonomic-routing', link: 'https://github.com/dev-academy-challenges/taxonomic-routing'},
```
Comment (bryceknz) [inline server/db/seeds/exercises.js]: Looks like there's a typo in the column name.

File: server/db/seeds/users.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
+exports.seed = function (knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('users').del()
+    .then(function () {
+      // Inserts seed entries
+      return knex('users').insert([
+        { id: 1, name: 'Jane', surname: 'Doe', email: 'janedoe@mail.com', hash: '#', role: 'facilitator' },
```
Comment (bryceknz) [inline server/db/seeds/users.js]: The fields should be 'firstName' and 'lastName'.

**Review by bryceknz — APPROVED** [review]

**Pull request merged**